include/modules/hyprland/workspaces.hpp

```diff
@@ -80,10 +80,13 @@ class Workspaces : public AModule, public EventHandler {
   void create_workspace(Json::Value& value);
   void remove_workspace(std::string name);
   void set_urgent_workspace(std::string windowaddress);
+  void parse_config(const Json::Value& config);
+  void register_ipc();
 
   bool all_outputs_ = false;
   bool show_special_ = false;
   bool active_only_ = false;
+  std::string sort_by = "default";
```

Comment:
We could instead use an enum and then parse the string into the enum in the constructor.

Reply:
Is there another example of a config value using an enum? From what I can see you have to implement mapping logic to convert a string to an enum in C++ since it's not an actual object.

Reply:
Yes, you write a function to map a string to the correct enum value. Implementing ostreams is also useful.